docs/avro.rst

```diff
@@ -1,98 +1,65 @@
+1. :ref:`BACKWARD compatibility <avro-backward_compatibility>`
+2. :ref:`FORWARD compatibility <avro-forward_compatibility>`
+3. :ref:`FULL compatibility <avro-full_compatibility>`
+4. :ref:`no compatibility checking <avro-none_compatibility>`
 
+Transitive
+^^^^^^^^^^
 
-Schema Evolution
-^^^^^^^^^^^^^^^^
+.. include:: includes/transitive.rst
 
-An important aspect of data management is **schema evolution**.  After the initial schema is defined, applications may
-need to evolve it over time. When this happens, it's critical for the downstream consumers to be able to handle data
-encoded with both the old and the new schema seamlessly. This is an area that tends to be overlooked in practice until
-you run into your first production issues.  Without thinking through data management and schema evolution carefully,
-people often pay a much higher cost later on.
+Transitive compatibility checking is important once you have more than two versions for a given subject.
+Refer to an `example of schema changes <https://github.com/confluentinc/schema-registry/issues/209>`__ which are incrementally compatible, but not transitively so.
 
-There are three common patterns of schema evolution:
 
-1. :ref:`backward compatibility <avro-backward_compatibility>`
-2. :ref:`forward compatibility <avro-forward_compatibility>`
-3. :ref:`full compatibility <avro-full_compatibility>`
 
+Compatibility Types
+-------------------
 
 .. _avro-backward_compatibility:
 
 Backward Compatibility
 ^^^^^^^^^^^^^^^^^^^^^^
 
-Backward compatibility means that data encoded with an older schema can be read with a newer schema.
+``BACKWARD`` compatibility means that consumers using the new schema can read data produced with the last schema.
+For example, if there are three schemas for a subject that change in order `X-2`, `X-1`, and `X` then ``BACKWARD`` compatibility ensures that consumers using the new schema `X` can process data written by producers using schema `X` or `X-1`, but not necessarily `X-2`.
+If the consumer using the new schema needs to be able to process data written by all registered schemas, not just the last two schemas, then use ``BACKWARD_TRANSITIVE`` instead of ``BACKWARD``.
+For example, if there are three schemas for a subject that change in order `X-2`, `X-1`, and `X` then ``BACKWARD_TRANSITIVE`` compatibility ensures that consumers using the new schema `X` can process data written by producers using schema `X`, `X-1`, or `X-2`.
+
+* ``BACKWARD``: consumer using schema `X` can process data produced with schema `X` or `X-1`
+* ``BACKWARD_TRANSITIVE``: consumer using schema `X` can process data produced with schema `X`, `X-1`, or `X-2`
+
+An example of a backward compatible change is a removal of a field. A consumer that was developed to process events without this field will be able to process events written with the old schema and contain the field – the consumer will just ignore that field.
```

Comment:
I would also add what backward compatibility doesn't mean: There is no guarantee that consumers using old schema (X-1) will be able to read new data (X). Which means that you need to finish upgrading all consumers before you start producing new events (unless your consumers are not processing data at all and you just copy the data to HDFS, per your example below).

Reply:
Ack!